utils/s3/client.cc

```diff
@@ -112,8 +129,13 @@ shared_ptr<client> client::make(std::string endpoint, endpoint_config_ptr cfg, s
 }
 
 future<> client::authorize(http::request& req) {
-    if (!_credentials || std::chrono::system_clock::now() >= _credentials.expires_at) {
-        _credentials = co_await _creds_provider_chain.get_aws_credentials();
+    if (!_credentials) {
```

Comment:
`[[unlikely]]` as this is expected not to be true ever, because creds are updated by timers